lib/mix/tasks/gen/ash.gen.expression.ex

```diff
@@ -0,0 +1,99 @@
+if Code.ensure_loaded?(Igniter) do
+  defmodule Mix.Tasks.Ash.Gen.Expression do
+    @moduledoc """
+    Generates a custom expression
+
+    See [Ash.CustomExpression](https://hexdocs.pm/ash/Ash.CustomExpression.html) for more.
+
+    ## Example
+
+    ```bash
+    mix ash.gen.expression MyApp.Expressions.LevenshteinDistance --args string,string
+    ```
+
+    ## Options
+
+    * `--args` - A comma separated list of argument types. i.e `--args string`
+    """
+    @shortdoc "Generates a custom expression module."
+    use Igniter.Mix.Task
+
+    @impl true
+    def info(_argv, _parent) do
+      %Igniter.Mix.Task.Info{
+        positional: [:expression],
+        schema: [args: :csv, name: :string]
+      }
+    end
+
+    @impl Igniter.Mix.Task
+    def igniter(igniter) do
+      expression = igniter.args.positional.expression
+      options = igniter.args.options
+      name = name(expression, options[:name])
+
+      args =
+        case options[:args] do
+          nil -> ""
+          args -> args |> Enum.map(&":#{&1}") |> Enum.join(",")
+        end
+
+      expression = Igniter.Project.Module.parse(expression)
+
+      igniter
+      |> Igniter.Project.Module.create_module(expression, """
+      use Ash.CustomExpression,
+        name: :#{name},
+        args: [#{args}]
+
+      @impl true
+      def expression(_data_layer, args) do
+        {:ok, expr(args)}
+      end
+      """)
+    end
+
+    defp name(_expression, name) when is_binary(name), do: name
+
+    defp name(expression, _name) do
```

Comment:
This seems hackish, I'm sorry. 😅 I'm hoping you know a better way. 🐛

Reply:
Seems like the right approach, but you can do the camelization to snake with `Macro.underscore`.

Reply:
That was exactly what I was looking for, big diff! 😂